TSRM/tsrm_win32.c

```diff
@@ -688,7 +688,20 @@ TSRM_API int shmget(key_t key, size_t size, int flags)
 		CloseHandle(shm_handle);
 		return -1;
 	}
-	shm->segment = shm_handle;
+	if (NULL == shm->segment || shm->segment == INVALID_HANDLE_VALUE) {
+		shm->segment = shm_handle;
+		shm->created = created;
+	}
+	else if (created == TRUE) {
+		CloseHandle(shm->segment);
+		shm->segment = shm_handle;
+		shm->created = TRUE;
+	}
+	else {
+		CloseHandle(shm_handle);
+	}
+	if(shm->descriptor==NULL)shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
```

Comment:
just to help before proper preview, I doubt that is your intentional intention (forgot to remove below line ?), see :

```c
if(shm->descriptor==NULL)shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);

shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
```

Reply:
Also little nit ; might be best to be consistent, style-wise maybe ?

```c
if(shm->descriptor==NULL) {
    shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
}
```

Reply:
> just to help before proper review, I doubt that is your intentional intention (forgot to remove below line ?), see :
> 
> ```c
> if(shm->descriptor==NULL)shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
> 
> shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
> ```

Yes, you write everything correctly. (forgot to remove below line)





> Also little nit ; might be best to be consistent, style-wise maybe ?
> 
> ```c
> if(shm->descriptor==NULL) {
>     shm->descriptor = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
> }
> ```

I don't mind.